cmd/sanitize-prow-jobs/main.go

```diff
@@ -1,27 +1,43 @@
 package main
 
 import (
+	"context"
 	"flag"
 	"fmt"
 	"io/ioutil"
 	"os"
+	"path"
 	"path/filepath"
 	"regexp"
 	"strings"
 	"sync"
+	"time"
 
 	"github.com/pkg/errors"
+	v1 "github.com/prometheus/client_golang/api/prometheus/v1"
 	"github.com/sirupsen/logrus"
 	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/sets"
 	prowconfig "k8s.io/test-infra/prow/config"
+	"k8s.io/test-infra/prow/logrusutil"
 	"sigs.k8s.io/yaml"
+
+	"github.com/openshift/ci-tools/pkg/dispatcher"
 )
 
 type options struct {
 	prowJobConfigDir string
 	configPath       string
 
 	help bool
+
+	prometheusURL          string
```

Comment:
IMO this would be better off in a standalone (simple) tool instead of adding to a mega-tool

Reply:
Ha. I actually started with a new tool, called `prow-job-dispatcher` but switched to this.
Maybe this mega-tool only do prometheus query? Nothing about prow?
Let me give it a try.

Reply:
Updated with a new tool.
Where should I put the logic for using job-volume-map?

We could use this tool `dispatch-prow-jobs` to generate the config for `sanitize-prow-jobs`. What you think?

Reply:
Oh, I see, since `sanitize` now sets the cluster ... yeah a two-step process could make sense. I don't think we should be calling a Prometheus API during `make jobs` for instance

Reply:
Result of slack discussion:
Do the Prom. query and use the answer in the new tool. The new tool will use the sanitize tool’s config too because it needs to know the jobs which have to be on some cluster. And then the new tool will dispatch others jobs and save it to sanitize’s config.